### PR TITLE
fix: no longer warns when no name was provided for global module

### PR DESCRIPTION
**Pull request by alan-agius4**
When the dictionary lookup or function look up return undefined or null, a warning should be emitted and the module name should be guessed.

If the user, doesn't receive a warning that the global name was not found, they won't be able to fix a possible issue. If the global name is incorrect or later set to null.

Closes: #2358

Some more context
https://github.com/dherges/ng-packagr/pull/1026#issuecomment-408605437

### CODE REVIEW

File: src/Chunk.ts [hunk 1]
```diff
@@ -68,8 +68,17 @@ function getGlobalName(
...
+	let globalName: string | undefined;
+	if (typeof globals === 'function') {
+		globalName = globals(module.id);
+	} else if (globals) {
+		globalName = globals[module.id];
+	}
+
+	if (globalName) {
+		return globalName;
+	}
```
Comment (guybedford) [inline src/Chunk.ts]: I think it's ok to not warn when the user has not specified any configuration, as in many cases this is the required behaviour that the package name be the request package name (eg this is the default on npm that a request package name and output package name can match).

The issue maybe arises in environments in which you are defining a global function or map that is trying to be authoritative and then has these invisible misses.

Instead, perhaps let's only warn for the case where `globals` is provided but has misses, by adding that as another case above?

Comment (alan-agius4) [reply src/Chunk.ts]: I am not 100% sure about this, as for UMD, I noticed that a lot of guessed name would be incorrect. If for instance having a `.` in the name. Or when the umd bundle name is totally different, and without a warning, it's hard to rollup users to know that something might be wrong.

However I can do what you suggested.

Comment (alan-agius4) [reply src/Chunk.ts]: @guybedford pushed an update.

File: src/Chunk.ts [hunk 2]
```diff
@@ -68,9 +68,18 @@ function getGlobalName(
...
+
+	if (hasExports || (hasExports ** globals && !globalName)) {
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply src/Chunk.ts]: yeah looks like the IDE didn't save :P

File: src/Chunk.ts [hunk 3]
```diff
@@ -68,9 +68,18 @@ function getGlobalName(
...
+		globalName = globals(module.id);
+	} else if (globals) {
+		globalName = globals[module.id];
+	}
+
+	if (globalName) {
+		return globalName;
+	}
+
+	if (hasExports || (hasExports && globals && !globalName)) {
```
Comment (guybedford) [inline src/Chunk.ts]: `if (hasExports || globals && !globalName)` would be simpler here I think.

**Comment by guybedford** [comment]
I've been thinking about this some more and maybe you are right we should always warn actually...

**Comment by alan-agius4** [comment]
hahah, removed the last commit :-)

**Comment by guybedford** [comment]
LGTM

**Pull request merged**